include/exadg/fluid_structure_interaction_precice/driver_fluid.h

```diff
@@ -0,0 +1,593 @@
+        get_dynamic_mapping<dim, Number>(this->application->get_grid_fluid(), fluid_grid_motion);
+      fluid_matrix_free->reinit(*mapping,
+                                fluid_matrix_free_data->get_dof_handler_vector(),
+                                fluid_matrix_free_data->get_constraint_vector(),
+                                fluid_matrix_free_data->get_quadrature_vector(),
+                                fluid_matrix_free_data->data);
+
+      // setup Navier-Stokes operator
+      fluid_operator->setup(fluid_matrix_free, fluid_matrix_free_data);
+
+      // setup postprocessor
+      fluid_postprocessor = this->application->create_postprocessor_fluid();
+      fluid_postprocessor->setup(*fluid_operator);
+
+      this->timer_tree.insert({"FSI", "Setup", "Fluid"}, timer_local.wall_time());
+    }
+    /****************************************** FLUID *******************************************/
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    // writing
+    this->precice =
+      std::make_shared<Adapter::Adapter<dim, dim, VectorType>>(this->precice_parameters);
+
+    write_mesh_name = this->precice_parameters.write_mesh_name;
+    write_data_name = this->precice_parameters.write_data_name;
+    this->precice->add_write_interface(this->application->get_boundary_descriptor_fluid()
+                                         ->velocity->dirichlet_mortar_bc.begin()
+                                         ->first,
+                                       this->precice_parameters.write_mesh_name,
+                                       {this->precice_parameters.write_data_name},
+                                       "values_on_quads",
+                                       fluid_matrix_free,
+                                       fluid_operator->get_dof_index_velocity(),
+                                       fluid_operator->get_quad_index_velocity_linear());
+
+    // structure to ALE
+    {
+      if(this->application->get_parameters_fluid().mesh_movement_type ==
+         IncNS::MeshMovementType::Poisson)
+      {
+        std::vector<unsigned int> quad_indices;
+        if(this->application->get_parameters_ale_poisson().spatial_discretization ==
+           Poisson::SpatialDiscretization::DG)
+          quad_indices.emplace_back(ale_poisson_operator->get_quad_index());
+        else if(this->application->get_parameters_ale_poisson().spatial_discretization ==
+                Poisson::SpatialDiscretization::CG)
+          quad_indices.emplace_back(ale_poisson_operator->get_quad_index_gauss_lobatto());
+        else
+          AssertThrow(false, ExcNotImplemented());
+
+        // VectorType stress_fluid;
+        communicator_ale = std::make_shared<InterfaceCoupling<dim, dim, Number>>(this->precice);
+        VectorType displacement_structure;
+        ale_poisson_operator->initialize_dof_vector(displacement_structure);
```

Comment:
is this variable used? same for `VectorType displacement_structure;` below and `VectorType velocity_structure;` below.